### PR TITLE
Allow observing only some blocks in ObserveFields

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

This is useful for visualizations, e.g. in a BNS simulation we only care about the stars, not the matter in the wave zone.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ParallelAlgorithms/Events/ObserveFields.hpp [hunk 1]
```diff
@@ -157,9 +158,19 @@ class ObserveFields<VolumeDim, tmpl::list<Tensors...>,
     using type = FloatingPointType;
   };
 
+  /// \brief A list of block or group names on which to observe.
+  ///
+  /// Set to `All` to observer everywhere.
```
Comment (kidder) [inline src/ParallelAlgorithms/Events/ObserveFields.hpp]: observe

**Review by kidder — APPROVED** [review]

**Review by kidder — CHANGES_REQUESTED** [review]
squash the requested change

**Comment by nilsdeppe** [comment]
Done and rebased. Thanks for the review!

**Review by nilsvu — DISMISSED** [review]
Nice!

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
Sorry, had to fix a silent conflict with include file names.

**Review by kidder — APPROVED** [review]

**Pull request merged**